src/jit/morph.cpp

```diff
@@ -13118,6 +13118,12 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                         tree->gtType = typ = temp->TypeGet();
                         foldAndReturnTemp  = true;
                     }
+                    else if (!varTypeIsStruct(typ) && (lvaTable[lclNum].lvType == typ) &&
+                             !lvaTable[lclNum].lvNormalizeOnLoad())
+                    {
```

Comment:
I think you shodul try to combine these two `else if `statements. and update the comment above sincewe now are handling non-short types.  

Is there a reason that you used an exact type test instead of the sanme sized test used by the other else if?
 i.e:  `(lvaTable[lclNum].lvType == typ)`'
vs.  `(genTypeSize(lvaTable[lclNum].lvType) == genTypeSize(typ))`

Reply:
Using the size only works on small int types, exactly because they're int. Otherwise you need to check for exact type to avoid int/float reinterpretation.